packages/nimbus/src/components/avatar/avatar.tsx

```diff
@@ -18,13 +21,32 @@ export const Avatar = (props: AvatarProps) => {
     ref,
     ...rest,
   };
+
+  const onLoad = () => {
+    setImageLoaded(true);
+    setImageError(false);
+  };
+
+  const onError = () => {
+    setImageLoaded(false);
+    setImageError(true);
+  };
+
+  // Show initials if no src provided, image hasn't loaded yet, or image failed
+  const shouldShowInitials = !src || !imageLoaded || imageError;
```

Comment:
While we wait for the image to load, we should show initials. The first and last names are required props anyway.

Reply:
my one concern about showing initials during load is that it might cause the initials to flash quickly, which could be jarring. it might be better to only display them if theres no src or an error?